### PR TITLE
RXX, RYY and RZZ gates

### PR DESCRIPTION
**Pull request by damarkian**
Added 2-qubit RXX, RYY and RZZ gates for #591 in https://github.com/qiboteam/qibo/commit/a75ab7a2c4e11d4aa56b793b85886cbaccce0020

### CODE REVIEW

**Comment by scarrazza** [comment]
@damarkian thanks for this. Could you please merge the latest master in this PR and fix conflicts?

File: src/qibo/gates/gates.py [hunk 1]
```diff
@@ -1049,6 +1055,117 @@ def parameters(self, x):
...
+        self._controlled_gate = None
+        self.target_qubits = (q0, q1)
+
+        self.parameters = theta
+        self.init_args = [q0, q1]
+        self.init_kwargs = {"theta": theta, "trainable": trainable}
+
+    def _dagger(self) -> "Gate":
+        """"""
+        return self.__class__(
```
Comment (scarrazza) [inline src/qibo/gates/gates.py:1082]: @damarkian thanks for this. Could you please include some test which covers this return statement?

**Review by scarrazza — APPROVED** [review]
LGTM, @stavros11 could you please have a look too?

**Comment by stavros11** [comment]
@damarkian thank you very much for implementing this. Everything looks good, only one last request: could you please add these gates in [`circuit.draw()`](https://github.com/qiboteam/qibo/blob/f639b7628f090ef46af2833c55dbc0b23a0c7410/src/qibo/models/circuit.py#L1289) in circuit.py? You just need to add a trivial mapping `{"rxx": "RXX", "ryy": "RYY", ...}` in the `labels` dictionary that is already there. 

If you like to test this you can use a script like this:
```py
from qibo import gates
from qibo.models import Circuit

c = Circuit(4)
c.add(gates.RXX(0, 1, theta=0.1))
c.add(gates.RYY(1, 2, theta=0.1))
c.add(gates.RZZ(0, 3, theta=0.1))
c.add(gates.RXX(3, 1, theta=0.1))

print(c.draw())
```

@scarrazza perhaps we should move these labels to the gates so that it is less confusing when someone adds a new gate.

**Comment by scarrazza** [comment]
@stavros yes, I fully agree.

**Review by stavros11 — APPROVED** [review]
Thank you, everything looks good to me.

**Pull request merged**